src/common/state/table.h

```diff
@@ -44,6 +44,11 @@ typedef struct {
   table_fail_callback fail_callback;
 } RetryInfo;
 
+static const RetryInfo heartbeat_retry = {
```

Comment:
Oh I see that you're using these retries as the actual mechanism for sending heartbeats.